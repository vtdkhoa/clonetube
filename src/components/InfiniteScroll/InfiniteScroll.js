import React, { Fragment } from 'react'
import { Waypoint } from 'react-waypoint'
import { Loader } from 'semantic-ui-react'
import PropTypes from 'prop-types'
import './InfiniteScroll.scss'

const InfiniteScroll = props => {
  return (
    <Fragment>
      {props.children}
      <Waypoint onEnter={props.bottomReachedCallback}>
        <div className="loader-container">
          <Loader active={props.showLoader} inline="centered"/>
        </div>
      </Waypoint>
    </Fragment>
  )
}

InfiniteScroll.propTypes = {
  bottomReachedCallback: PropTypes.func,
  showLoader: PropTypes.bool
}

export default InfiniteScroll