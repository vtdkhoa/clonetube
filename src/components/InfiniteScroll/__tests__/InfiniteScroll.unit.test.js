import React from 'react'
import InfiniteScroll from '../InfiniteScroll'
import { shallow } from 'enzyme'

describe('InfiniteScroll', () => {
  test('renders empty children', () => {
    const wrapper = shallow(<InfiniteScroll/>)
    expect(wrapper).toMatchSnapshot()
  })

  test('renders with chilren', () => {
    const wrapper = shallow(
      <InfiniteScroll>
        <div>Children props</div>
      </InfiniteScroll>
    )
    expect(wrapper).toMatchSnapshot()
  })
})