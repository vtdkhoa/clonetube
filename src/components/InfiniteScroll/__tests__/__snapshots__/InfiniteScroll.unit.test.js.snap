// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfiniteScroll renders empty children 1`] = `
<Fragment>
  <Waypoint
    bottomOffset="0px"
    debug={false}
    fireOnRapidScroll={true}
    horizontal={false}
    onEnter={[Function]}
    onLeave={[Function]}
    onPositionChange={[Function]}
    topOffset="0px"
  >
    <div
      className="loader-container"
    >
      <Loader
        inline="centered"
      />
    </div>
  </Waypoint>
</Fragment>
`;

exports[`InfiniteScroll renders with chilren 1`] = `
<Fragment>
  <div>
    Children props
  </div>
  <Waypoint
    bottomOffset="0px"
    debug={false}
    fireOnRapidScroll={true}
    horizontal={false}
    onEnter={[Function]}
    onLeave={[Function]}
    onPositionChange={[Function]}
    topOffset="0px"
  >
    <div
      className="loader-container"
    >
      <Loader
        inline="centered"
      />
    </div>
  </Waypoint>
</Fragment>
`;
